BE/src/script_gen/nodes/yt_fetcher.py

```
@@ -198,7 +211,43 @@ def _search_related_videos_for_display(keywords: List[str]) -> List[Dict]:
+                    if duration_sec < 180:
+                        continue
+                    view_count = int(stats.get("viewCount", 0))
+                    published_at = snippet.get("publishedAt", "")
+                    velocity = 0.0
+                    if published_at:
+                        try:
+                            pub_date = datetime.fromisoformat(published_at.replace("Z", "+00:00"))
+                            days = max((datetime.now(timezone.utc) - pub_date).days, 1)
+                            velocity = view_count / days
+                        except Exception:
+                            pass
```

Ruff (0.15.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)